network/address_test.go

```diff
@@ -27,6 +29,12 @@ func TestConnType(t *testing.T) {
 }
 
 func TestAddress(t *testing.T) {
+	localhostNetworkAddresses, _ := net.LookupHost("localhost")
+	sort.Strings(localhostNetworkAddresses)
+	facebookNetworkAddresses, _ := net.LookupHost("facebook.com")
+	sort.Strings(facebookNetworkAddresses)
+	googleNetworkAddresses, _ := net.LookupHost("google.com")
```

Comment:
I don't care for debugging-variables. You can even make them private, then as long as you have a unit-test, you can overwrite it.

Again, I would not define the interface globally, but the lookup-function. So

```
var lookupHost func(host string)(addrs []string, error) = net.LookupHost
```

then you can overwrite that for the tests.